commit 07db5bd23571ee9f3da518f81b03a8eb3c44fc6c
Author: Alan Previn <alan.previn.teres.alexis@intel.com>
Date:   Mon Nov 7 20:56:28 2022 -0800

    drm/i915/pxp: Separate PXP FW interface structures for both v42 and 43
    
    Previously, we only used PXP FW interface version-42 structures for
    PXP arbitration session on ADL/TGL products and version-43 for HuC
    authentication on DG2. That worked fine despite not differentiating such
    versioning of the PXP firmware interaction structures. This was okay
    back then because the only commands used via version 42 was not
    used via version 43 and vice versa.
    
    With MTL, we'll need both these versions side by side for the same
    commands (PXP-session) with the older platform feature support. That
    said, let's create separate files to define the structures and definitions
    for both version-42 and 43 of PXP FW interfaces.
    
    Signed-off-by: Alan Previn <alan.previn.teres.alexis@intel.com>
    Reviewed-by: Daniele Ceraolo Spurio <daniele.ceraolospurio@intel.com>
    Signed-off-by: Daniele Ceraolo Spurio <daniele.ceraolospurio@intel.com>
    Link: https://patchwork.freedesktop.org/patch/msgid/20221108045628.4187260-2-alan.previn.teres.alexis@intel.com

diff --git a/drivers/gpu/drm/i915/pxp/intel_pxp_cmd_interface_43.h b/drivers/gpu/drm/i915/pxp/intel_pxp_cmd_interface_43.h
new file mode 100644
index 000000000000..ad67e3f49c20
--- /dev/null
+++ b/drivers/gpu/drm/i915/pxp/intel_pxp_cmd_interface_43.h
@@ -0,0 +1,26 @@
+/* SPDX-License-Identifier: MIT */
+/*
+ * Copyright(c) 2022, Intel Corporation. All rights reserved.
+ */
+
+#ifndef __INTEL_PXP_FW_INTERFACE_43_H__
+#define __INTEL_PXP_FW_INTERFACE_43_H__
+
+#include <linux/types.h>
+#include "intel_pxp_cmd_interface_cmn.h"
+
+/* PXP-Cmd-Op definitions */
+#define PXP43_CMDID_START_HUC_AUTH 0x0000003A
+
+/* PXP-Input-Packet: HUC-Authentication */
+struct pxp43_start_huc_auth_in {
+	struct pxp_cmd_header header;
+	__le64 huc_base_address;
+} __packed;
+
+/* PXP-Output-Packet: HUC-Authentication */
+struct pxp43_start_huc_auth_out {
+	struct pxp_cmd_header header;
+} __packed;
+
+#endif /* __INTEL_PXP_FW_INTERFACE_43_H__ */